File: Data Structure/Binary Tree/LET_no1302_Deepest Leaves Sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def deepestLeavesSum(self, root: TreeNode) -> int:
        memo = collections.defaultdict(int)
        self.dfs(root, 1, memo)
        return memo[max(memo)]

    def dfs(self, root, level, memo):
        if not root:
            return
        memo[level] += root.val
        self.dfs(root.left, level + 1, memo)
        self.dfs(root.right, level + 1, memo)

File: Breadth-First Search/Binary tree/LET_no1302_Deepest Leaves Sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int deepestLeavesSum(TreeNode* root) {
        int res = 0, i;
        queue<TreeNode*> q;
        q.push(root);
        
        while (!q.empty()) {
            for (i = q.size() - 1, res = 0; i >= 0; i--) {
                TreeNode* cur = q.front();
                q.pop();
                res += cur->val;
                if (cur->left) q.push(cur->left);
                if (cur->right) q.push(cur->right);
            }
        }
        
        return res;
    }
};